src/main/java/bennett/citibike/json/CitiBikeUtils.java

```diff
@@ -0,0 +1,87 @@
+package bennett.citibike.json;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+
+public class CitiBikeUtils {
+
+    // Find the status of a station given the station_id
+    public Optional<Status> findStationStatus(String stationId, List<Status> statusList) {
+        return statusList.stream()
+                .filter(status -> status.stationId.equals(stationId))
+                .findFirst();
+    }
+
+    // Find the closest station (with available bikes) to a given location
+    public Optional<Station> findClosestStationWithBikes(double lat,
+                                                         double lon,
+                                                         List<Station> stations,
+                                                         List<Status> statusList) {
+        return findClosestStation(lat, lon, stations, statusList, true);
+    }
+
+    private Optional<Station> findClosestStation(double lat,
+                                                 double lon,
+                                                 List<Station> stations,
+                                                 List<Status> statusList,
+                                                 boolean isBikeSearch) {
+        return stations.stream()
+                .filter(station -> {
+                    Optional<Status> stationStatus = findStationStatus(
+                            station.stationId, statusList);
+                    return stationStatus.isPresent()
+                            && (isBikeSearch ? stationStatus.get().availableBikes > 0 :
+                                    stationStatus.get().availableSlots > 0);
+                })
+                .min(Comparator.comparingDouble(station -> calculateDistance(lat, lon,
+                        station.lat, station.lon)));
+    }
+
+    public static double calculateDistance(double lat1, double lon1, double lat2, double lon2) {
```

Comment:
While I understand how this is appropriate, as I said before, if you write `static` methods I fail you.